test/tests/functional/pbs_pbsnodes_output_trimmed.py

```diff
@@ -70,6 +70,8 @@ def test_pbsnodes_output(self):
         hname = "long123456789012345678901234567890.pbspro.com"
         ret = self.hostname_resolves(hname)
         if ret == 0:
+            if os.getuid() != 0:
```

Comment:
shouldn't this be the first thing that you do in this test? why is this check so far down?

Reply:
@agrawalravi90 the test will pass if hostname is resolvable and will try to edit /etc/hosts file in case it is not hence we do not want to skip at first place.

Reply:
Thanks for your reply Anamika. The possibility of there being an actual host with that ridiculously long and made up hostname ("long123456789012345678901234567890.pbspro.com") is negligible. So, I think we should make the test less confusing by just checking for root permission at the top and just editing /etc/hosts file without checking for the hostname to be resolvable. Even if there are multiple entries in /etc/hosts (just in case that hostname actually exists), the test should work fine, the resolvers will just pick the first entry in /etc/hosts for that hostname

Reply:
@agrawalravi90 What you are saying makes sense, except that for the internal testbed systems this absurd hostname has already been added for testing purposes. The reason for adding was to avoid editing an important file such as  /etc/hosts file. Also, if we write the skip code at the beginning of the test like you suggested, we wouldn't be able to run it as pbsroot user.

Reply:
wow, ok. If this test exists to just check that pbsnodes output doesn't trim a really long vnode name, then how about instead of editing the /etc/hosts file, we edit the vnode def file to create a vnode with that ridiculous name with the existing mom? that won't require this test to run as root at all.

Reply:
and we can rename that poor host in our testbed systems

Reply:
@agrawalravi90 that is a good suggestion. @VishrutArora please try it out.

Reply:
If all you want is a very long vnode name, then you can create one.  You say qmgr -c 'c n <very long name> mom=hostname'.  The vnode will immediately go stale, but it will exist for the purposes of using pbsnodes.